service/controller/v2/resource/deployment/deployment.go

```diff
@@ -70,27 +71,38 @@ func (r Resource) newMainDeployment(obj providerv1alpha1.AzureConfig) (deploymen
 		"templatesBaseURI":              baseTemplateURI(r.templateVersion),
 	}
 
-	d := deployment{
-		Name:                   mainDeploymentName,
-		Parameters:             convertParameters(params),
-		ResourceGroup:          key.ClusterID(obj),
-		TemplateURI:            templateURI(r.templateVersion, mainTemplate),
-		TemplateContentVersion: templateContentVersion,
+	d := azureresource.Deployment{
+		Properties: &azureresource.DeploymentProperties{
+			Mode:       azureresource.Incremental,
+			Parameters: convertParameters(defaultParams, overwrites),
+			TemplateLink: &azureresource.TemplateLink{
+				URI:            to.StringPtr(templateURI(r.templateVersion, mainTemplate)),
+				ContentVersion: to.StringPtr(templateContentVersion),
+			},
+		},
 	}
 
 	return d, nil
 }
 
-// convertParameters converts the map into the structure used by the Azure API.
-func convertParameters(inputs map[string]interface{}) map[string]interface{} {
-	params := make(map[string]interface{}, len(inputs))
-	for key, val := range inputs {
-		params[key] = struct {
-			Value interface{}
-		}{
-			Value: val,
+// convertParameters merges the input maps and converts teh result into the
```

Comment:
Nitpick: Typo `teh` -> `the`. I'm not native English speaker and not that good in grammar, but I'm wondering if `converts the result into structure used by Azure API` would be more correct. Somehow these multiple `the` close to each other doesn't sound right. I might be totally wrong here though :)

Reply:
Fixed the typo and kept the articles. Based in my language understanding this makes sense.